var/spack/repos/builtin/packages/spectre/package.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Spectre(CMakePackage):
+    """SpECTRE is an open-source code for multi-scale, multi-physics problems in
+    astrophysics and gravitational physics. In the future, we hope that it can
+    be applied to problems across discipline boundaries in fluid dynamics,
+    geoscience, plasma physics, nuclear physics, and engineering. It runs at
+    petascale and is designed for future exascale computers.
+
+    SpECTRE is being developed in support of our collaborative Simulating
+    eXtreme Spacetimes (SXS) research program into the multi-messenger
+    astrophysics of neutron star mergers, core-collapse supernovae, and
+    gamma-ray bursts."""
+
+    homepage = "https://spectre-code.com"
+    url = "https://github.com/sxs-collaboration/spectre/archive/v2021.12.15.tar.gz"
+    git = "https://github.com/sxs-collaboration/spectre.git"
+
+    maintainers = ['nilsleiffischer']
+
+    generator = 'Ninja'
+
+    version('develop', branch='develop')
+    version('2022.01.03', sha256='872a0d152c19864ad543ddcc585ce30baaad4185c2617c13463d780175cbde5f')
+    version('2021.12.15', sha256='4bfe9e27412e263ffdc6fcfcb84011f16d34a9fdd633ad7fc84a34c898f67e5c')
+
+    # Configuration variants
+    variant('build_type',
+            default='Release',
+            values=("Release", "Debug", "RelWithDebInfo"),
+            description='CMake build type')
+    variant('executables',
+            values=any_combination_of(
+                'SolvePoisson1D', 'SolvePoisson2D', 'SolvePoisson3D',
+                'SolveElasticity2D', 'SolveElasticity3D', 'SolveXcts',
+            ),
+            description="Executables to install")
+    variant('python', default=False, description="Build Python bindings")
+    variant('doc', default=False, description="Build documentation")
+    # TODO: support installation of executables with shared libs
+    # variant('shared',
+    #         default=False,
+    #         description="Build shared libraries instead of static")
+    variant('pch',
+            default=True,
```

Comment:
If `pch` is broken, then it shouldn't be the default.

Reply:
Changed the default for now. PCH builds work on our CI (we always build with PCH because it speeds things up so much), only with Spack do I get `clang: error: cannot specify -o when generating multiple output files`. Trying to figure it out..